components/movementsensor/adxl345/adxl345.go

```diff
@@ -334,21 +339,36 @@ func makeAdxl345(
 		if err != nil {
 			return nil, err
 		}
+
+		if newConf.SingleTap.DebounceMs > 0 {
+			sensor.debounceMs = newConf.SingleTap.DebounceMs
+		} else {
+			// default is 3ms
+			sensor.debounceMs = 3
+		}
+
 		sensor.startInterruptMonitoring(ticksChan)
 	}
 
 	return sensor, nil
 }
 
 func (adxl *adxl345) startInterruptMonitoring(ticksChan chan board.Tick) {
+	debounced := debounce.New(time.Millisecond * time.Duration(adxl.debounceMs))
 	adxl.workers.AddWorkers(func(cancelContext context.Context) {
 		for {
 			select {
 			case <-cancelContext.Done():
 				return
 			case tick := <-ticksChan:
 				if tick.High {
-					utils.UncheckedError(adxl.readInterrupts(cancelContext))
+					if tick.Name == adxl.tapInterruptName {
+						// debounce on tap interrupt
+						debounced(func() { utils.UncheckedError(adxl.readInterrupts(cancelContext)) })
```

Comment:
If I read [the debounced code](https://github.com/bep/debounce/blob/master/debounce.go) correctly, this will call `readInterrupts` only if we haven't had any additional interrupts in the past `debounceMS` milliseconds. If, instead, we get lots of taps very quickly, it will never call `readInterrupts` because there are so many of them coming in (and `debounced` waits until they stop arriving). Is that the right behavior? My first instinct is to react immediately to the first tap, but then ignore the subsequent ones until they stop for long enough (whereas the `debounce` library was intended to react when someone stops typing in a text field and the UI should update).

There might be an easy way to adapt this to be "handle the interrupt right away if it's the first one we've seen for a while, and then ignore any that come in for the next `debounceMs`," but I don't see it yet. The `debounced` library is straightforward enough that we could probably just implement that behavior ourselves, if it's too hard to adapt (if that's the right intended behavior at all, and I might be wrong about that).

Reply:
This seems like a bit more than I'd want for a debounce time for something that reports that it can detect taps on its own. Then I thought - did we check the[ adxl data sheet](https://www.analog.com/media/en/technical-documentation/data-sheets/adxl345.pdf) for this functionality?

When looking at the datasheet I found:
![Screenshot 2024-08-02 at 14 52 45](https://github.com/user-attachments/assets/a2c76c14-9d33-4edc-92f1-7f556ca9c4fd)

So I'm leaning towards actually just logging an Info message and telling the user to experiment with `dur` and `threshold`.  cc @oliviamiller @cloudui